188_1_efektyvusPaskirstyk.cpp

```diff
@@ -0,0 +1,58 @@
+const int MAXN = ...;      // maksimalus knygų skaičius
+const int MAXK = ...;      // maksimalus darbuotojų skaičius
+const int BEGALINIS = ...; // kažkoks kuo didesnis skaičius, pavyzdžiui, 1e9
+
+int k;
+int n;
+int p; // puslapių skaičius
+int ivertis;
+int nuo[MAXN+MAXK+1];
+
+void paskristyk () {
+    int D[MAXK+1][MAXN+1], M[MAXK+1][MAXN+1];
+    int r[MAXN+MAXK+1]; // pagalbinis masyvas
+
+    // užpildomas masyvas r                              //**

```

Comment:
Kas čia per žvaigždutės gale eilutės? 😁

Reply:
Čia atseit yra du to paties uždavinio kodai ir šitas efektyvus, nes sumą skaičiuoja per O(1), kadangi naudoja pref sumas (tas "r" masyvas), o ankstesnis kodas - per O(n) skaičiuodavo, nes visad iš naujo imdavo. Tai knygoje ten paminėjo, kad žvaigždutės gale eilutės reikš, kad tai yra tos vietos, kur skiriasi tie du kodai.